src/ast/passes/semantic_analyser.cpp

```diff
@@ -1715,6 +1715,46 @@ void SemanticAnalyser::validate_map_key(const SizedType &key, Node &node)
   }
 }
 
+void SemanticAnalyser::visit(MapDeclStatement &decl)
+{
+  if (!bpftrace_.config_->get(ConfigKeyBool::unstable_map_decl)) {
+    decl.addError() << "Map declarations are not enabled by default. To enable "
+                       "this unstable feature, set this config flag to true "
+                       "e.g. unstable_map_decl=true";
```

Comment:
Nit: maybe the second sentence here is actually a `HINT`?

Reply:
Hmm... It feels like all one message to me.

Reply:
We don't do this and it doesn't matter to me, but the reason I'm calling out the distinction is that often you have an error, and then a richer diagnostic that has a recommended fix that you can apply. (Like we apply the `clang-tidy` fixes, which are even more structured.) This is definitely fitting that pattern, which is why it's already two separate sentences. I would do it just to set a good example for others since everything gets cargo-culted, and this structure generally makes this more testable and stable (because the tests don't need to overfit on the hint text!), e.g.

```
auto &err = decl.addError();
err << "Map declarations are not enabled by default.";
err.addHint() << "To enable this feature, set the config flag unstable_map_decl=true.";
```

Reply:
This is another hill I don't wish to die on but IMHO this isn't really an "error" or an indication that the user did something wrong but rather a notification that the feature they are using requires an explicit opt-in. The error message doesn't exist without the information on how to opt-in, which is why it isn't a hint.

I think we have plenty of examples in the code base of proper `addHint` usage.